### PR TITLE
BaSP-TG-12: TimeSheet (GetById, DeleteById and EditById)

### PR DESCRIPTION
**Pull request by NicolasGulam**
Created for TimeSheet Section Linking in Index file, Model, Validations, Routes and Functions GetById, DeleteById and EditById.

### CODE REVIEW

File: src/controllers/timeSheets.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import TimeSheets from '../models/TimeSheets';
+
+// Get by Id
```
Comment (ignacioaristo) [inline src/controllers/timeSheets.js:3]: This comment is redundant, with the name of the function below we know it.

File: src/controllers/timeSheets.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+const getTimeSheetById = async (req, res) => {
+  try {
+    const searchId = req.params.id;
+    const timeSheets = await TimeSheets.findById({ _id: searchId });
+
+    return res.status(200).json({
+      message: 'TimeSheet found',
+      data: timeSheets,
+      error: false,
```
Comment (ignacioaristo) [inline src/controllers/timeSheets.js:13]: i think this prop is not necessary.

File: src/controllers/timeSheets.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    });
+  } catch (error) {
+    return res.json({
+      message: 'An error ocurred',
```
Comment (ignacioaristo) [inline src/controllers/timeSheets.js:17]: we can handle the error message, showing the error in this line.

File: src/controllers/timeSheets.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      error: true,
+    });
+  }
+};
+
+// Delete by Id
```
Comment (ignacioaristo) [inline src/controllers/timeSheets.js:23]: same with this comment.

File: src/controllers/timeSheets.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+const deleteTimeSheet = async (req, res) => {
+  try {
+    const searchId = req.params.id;
+    const result = await TimeSheets.findByIdAndDelete({ _id: searchId });
+
+    return res.status(200).json({
+      message: `TimeSheet with the id ${searchId} deleted`,
+      data: result,
+      error: false,
```
Comment (ignacioaristo) [inline src/controllers/timeSheets.js:33]: same here., remove all

File: src/validations/timeSheets.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    date: Joi.date().required(),
+    task: Joi.string().valid('BE', 'FE').required(),
+  });
+
+  const validation = timeSheetValidation.validate(req.body);
+
+  if (validation.error) {
+    return res.status(400).json({
+      message: `There was an error: ${validation.error.datails[0].message}`,
+      data: undefined,
```
Comment (ignacioaristo) [inline src/validations/timeSheets.js:15]: i think this prop is not necessary, as the line below.

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/validations/timeSheets.js (hunk 6 above)
Comment (NicolasGulam) [reply src/validations/timeSheets.js]: All corrections have been made

File: src/data/time-sheets.json [hunk 7]
```diff
@@ -1,26 +1,26 @@
 [{
```
Comment (ignacioaristo) [inline src/data/time-sheets.json:1]: Delete this file, now, we are using the data base.

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/data/time-sheets.json (hunk 7 above)
Comment (NicolasGulam) [reply src/data/time-sheets.json]: Done.

Comment (NicolasGulam) [reply src/data/time-sheets.json]: I saw that i have new conflicts to resolve

**Review by ignacioaristo — APPROVED** [review]

**Review by ignacioaristo — DISMISSED** [review]

**Review by ignacioaristo — APPROVED** [review]

**Pull request merged**